_posts/2018-12-28-Reading-Web-Component-And-Custom-Elements.markdown

```diff
@@ -0,0 +1,26 @@
+---
+layout: post
+title:  "Reading: Make a Native Web Component with Custom Elements v1 and Shadow DOM v1"
+date: "2018-12-28 22:14:00 +0800"
+categories: "reading note"
+author: "Jackson"
+tags: "web component"
+---
+
+[Article Link](https://bendyworks.com/blog/native-web-components)   
+[中文版本](https://zhuanlan.zhihu.com/p/52802048)
+
+#### Summary:
+This article is mainly talking about the web component standards which includes the HTML Templates, Shadow Dom and Custom Elements.
+
+The author made a form component base on the web component's standards, so this component just like native HTML elements, they can have properties, methods, and event listeners. Just the different with React component, web component didn't need any translation engine to translate the code to standard HTML, It just made by vanilla JavaScript and the native HTML. So it doesn't need any dependency. And it will work across modern browsers and it can be used with any JavaScript library or framework that works with HTML. And with the advantage of React component reuse and share easily. And the custom element has lifecycle hooks too, so you can control the component easily.
```

Comment:
base on => based on
so this component "is" just like native HTML elements":" they can have...
Just the different ... => "The difference between web components and React component is that web components are just implemented by vanilla JavaScript and native HTML, so they don't need any transpilers to translate the code to standard HTML. So they don't need any extra dependency and will work across modern browsers. They can also be uased with any Javascript libraries or frameworks that works with standard HTML."
And with the advantage of React... (I am not exactly sure what you mean by this, but i guess it's like this) => Web components also have the same advantages of React components: reused and shared easily. And as custom elements have lifecycle hooks like React components, you can control the components easily.